src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -0,0 +1,213 @@
+                        ProcessSection(provider, nameof(AzureBlobEgressProviderOptions.AccountKey), includeChildSections: false, redact: true);
+                        ProcessSection(provider, nameof(AzureBlobEgressFactory.ConfigurationOptions.SharedAccessSignatureName), includeChildSections: false, redact: false);
+                        ProcessSection(provider, nameof(AzureBlobEgressFactory.ConfigurationOptions.AccountKeyName), includeChildSections: false, redact: false);
+                    }
+                    //Matches filesystem provider type.
+                    else if (egressProperties.Any(child => matchesProvider(child, EgressProviderTypes.FileSystem)))
+                    {
+                        ProcessSection(provider, nameof(EgressConfigureOptions.CommonEgressProviderOptions.Type), includeChildSections: false);
+                        ProcessSection(provider, nameof(FileSystemEgressProviderOptions.DirectoryPath), includeChildSections: false);
+                        ProcessSection(provider, nameof(FileSystemEgressProviderOptions.IntermediateDirectoryPath), includeChildSections: false);
+                        ProcessSection(provider, nameof(FileSystemEgressProviderOptions.CopyBufferSize), includeChildSections: false);
+                    }
+                    else
+                    {
+                        //Emit other egress entries, with redaction
+                        foreach (IConfigurationSection providerProperty in egressProperties)
+                        {
+                            if (string.Equals(providerProperty.Key, nameof(EgressConfigureOptions.CommonEgressProviderOptions.Type), StringComparison.OrdinalIgnoreCase))
+                            {
+                                ProcessSection(providerProperty, includeChildSections: false, redact: false);
+                            }
+                            else
+                            {
+                                ProcessSection(providerProperty, includeChildSections: true, redact: true);
+                            }
+                        }
+                    }
+                    _writer.WriteEndObject();
+                }
+                _writer.WriteEndObject();
+            }
+        }
+
+        private IConfigurationSection ProcessSection(IConfiguration parentSection, string key, bool includeChildSections = true, bool redact = false)
+        {
+            IConfigurationSection section = parentSection.GetSection(key);
+            if (!section.Exists())
+            {
+                _writer.WritePropertyName(key);
+                _writer.WriteStringValue("NOT PRESENT");
+                return null;
+            }
+
+            ProcessSection(section, includeChildSections, redact);
+
+            return section;
+        }
+
+        private void ProcessSection(IConfigurationSection section, bool includeChildSections = true, bool redact = false)
```

Comment:
This looks like it could write a property name without writing a value for it if (1) the section doesn't have any children but it was told include the children, or (2) not to include children but does have children.

Reply:
Will add comments.